opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/AppResources.java

```diff
@@ -222,86 +344,166 @@ else if (appStat.getPid() != null && appStat.getPid() != cli.get().getPid())
 					String.format("Event socket to %s closed by app", appStat.getAppId()), ex);
 		}
 	}
-	
+
 	@POST
 	@Path("appstart")
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_ADMIN, AuthorizationCheck.ODCS_API_USER})
 	public Response postAppStart(@QueryParam("appid") Long appId)
-		throws WebAppException, DbException, SQLException
+			throws WebAppException, DbException
 	{
-		LOGGER.debug("Post for appstart received with appId={}", appId);
 		if (appId == null)
-			throw new WebAppException(ErrorCodes.MISSING_ID, "appId parameter required for this operation.");
-		
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+		{
+			throw new WebAppException(HttpServletResponse.SC_BAD_REQUEST,
+					"appId parameter required for this operation.");
+		}
+
+		try (LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
 		{
 			// Retrieve ApiLoadingApp and ApiAppStatus
-			ApiLoadingApp loadingApp = dao.getApp(appId);
-			ApiAppStatus appStat = dao.getAppStatus(appId);
-			
+			ApiLoadingApp loadingApp = mapLoading(dai.getComputationApp(DbKey.createDbKey(appId)));
+			ApiAppStatus appStat = getAppStatus(dai, appId);
+
 			// Error if already running and heartbeat is current
-			if (appStat.getPid() != null && appStat.getHeartbeat() != null
-			&& (System.currentTimeMillis() - appStat.getHeartbeat().getTime() < 20000L))
+			if (appStat != null && appStat.getPid() != null && appStat.getHeartbeat() != null
+					&& (System.currentTimeMillis() - appStat.getHeartbeat().getTime() < 20000L))
 				throw new WebAppException(ErrorCodes.NOT_ALLOWED,
-					"App id=" + appId + " (" + loadingApp.getAppName() + ") is already running.");
-			
+						"App id=" + appId + " (" + loadingApp.getAppName() + ") is already running.");
+
 			// Error if no "startCmd" property
 			String startCmd = ApiPropertiesUtil.getIgnoreCase(loadingApp.getProperties(), "startCmd");
 			if (startCmd == null)
 				throw new WebAppException(ErrorCodes.BAD_CONFIG,
-					"App id=" + appId + " (" + loadingApp.getAppName() + ") has no 'startCmd' property.");
+						"App id=" + appId + " (" + loadingApp.getAppName() + ") has no 'startCmd' property.");
 
 			// ProcWaiterThread runBackground to execute command, use callback.
 			ProcWaiterCallback pwcb = (procName, obj, exitStatus) ->
-					{
-						ApiLoadingApp loadingApp1 = (ApiLoadingApp)obj;
-						LOGGER.info("App Termination: app {} was terminated with exit status {}",
-								loadingApp1.getAppName(), exitStatus);
-					};
+			{
+				ApiLoadingApp loadingApp1 = (ApiLoadingApp)obj;
+			};
```

Comment:
I don't understand what this accomplishes.

Reply:
Looks like this was just used for logging when an application fails. I removed the usage